control-plane/pkg/reconciler/broker/broker_test.go

```diff
@@ -2085,6 +2189,22 @@ func useTable(t *testing.T, table TableTest, env *config.Env) {
 			expectedTopicDetail = td.(sarama.TopicDetail)
 		}
 
+		externalTopicUsage := false
+		expectedTopicName := fmt.Sprintf("%s%s-%s", TopicPrefix, BrokerNamespace, BrokerName)
+		if _, ok := row.OtherTestData[externalTopic]; ok {
+			expectedTopicName = ExternalTopicName
+			externalTopicUsage = true
+		}
+
+		var metadata []*sarama.TopicMetadata
+		if externalTopicUsage {
+			metadata = append(metadata, &sarama.TopicMetadata{
+				Name:       ExternalTopicName,
+				IsInternal: false,
+				Partitions: []*sarama.PartitionMetadata{{}},
+			})
+		}
```

Comment:
why is this special case needed?
It seems that these metadata are good in both cases.